libraries/shared/watcher/storage_watcher_test.go

```diff
@@ -93,16 +129,119 @@ var _ = Describe("Storage Watcher", func() {
 				}
 				diffs = append(diffs, diff)
 			}
-			mockDiffsRepository.GetNewDiffsDiffs = diffs
-			mockDiffsRepository.GetNewDiffsErrors = []error{nil, fakes.FakeError}
+			setDiffsToReturn(storageWatcher.DiffStatus, mockDiffsRepository, diffs)
+			setGetDiffsErrors(storageWatcher.DiffStatus, mockDiffsRepository, []error{nil, fakes.FakeError})
 
 			err := storageWatcher.Execute()
 
 			Expect(err).To(HaveOccurred())
-			Expect(mockDiffsRepository.GetNewDiffsPassedMinIDs).To(ConsistOf(0, diffID))
+			Expect(err).To(MatchError(fakes.FakeError))
+			assertGetDiffsMinIDs(storageWatcher.DiffStatus, mockDiffsRepository, []int{0, diffID})
 		})
+	})
 
-		It("resets min ID to zero when previous query returns fewer than max results", func() {
+	It("resets min ID to zero when previous query returns fewer than max results", func() {
```

Comment:
Not sure if I'm parsing this correctly but it looks like this `It` statement is outside of the `Describe` for `Execute`?

Reply:
nice catch!